superset/models/core.py

```diff
@@ -636,14 +636,24 @@ def mutate_sql_based_on_config(self, sql_: str, is_split: bool = False) -> str:
             )
         return sql_
 
-    def get_df(  # pylint: disable=too-many-locals
+    def render_sql(self, sql: str, **kwargs: Any) -> str:
+        tp = jinja_context.get_template_processor(self)
+        return tp.process_template(sql, **kwargs)
+
+    def get_result_set(
```

Comment:
Factoring `get_result_set` out of  `get_df` as its needed places, and makes the method a bit more pallatable.